Features/Context/SubContext/CommonActions.php

```diff
@@ -10,10 +10,29 @@
 namespace EzSystems\PlatformUIBundle\Features\Context\SubContext;
 
 use EzSystems\BehatBundle\Helper\EzAssertion;
+use Behat\Behat\Hook\Scope\BeforeScenarioScope;
 
 trait CommonActions
 {
     /**
+     * @var EzSystems\BehatBundle\Context\Object\BasicContent
+     */
+    private $basicContentContext;
+
+    /** @BeforeScenario */
+    public function gatherContexts(BeforeScenarioScope $scope)
+    {
+        // not all context need the BasicContent Context
+        $this->basicContentContext = null;
+
+        $environment = $scope->getEnvironment();
+        $basicContextNamespace = 'EzSystems\BehatBundle\Context\Object\BasicContent';
+        if ($environment->hasContextClass($basicContextNamespace)) {
+            $this->basicContentContext = $environment->getContext($basicContextNamespace);
+        }
```

Comment:
If I understand correctly, this looks OK but what happens if/when more contexts are needed (even if on instances outside the trait)?

I would perhaps refactor it to hold an indexed array of contexts (`private $contextInstances`).

OTOH, the contexts could be fetched directly from the environment, which already has the logic for checking and exception handling (see https://github.com/Behat/Behat/blob/master/src/Behat/Behat/Context/Environment/InitializedContextEnvironment.php#L110 )

I just find odd that InitializedContextEnvironment does not return and indexed array of contexts (only keys OR values), but in any case the `getContext($class)` function should suffice.

To sum, I would suggest:
1. in `gatherContexts()`, storing either the indexed contexts or the environment itself.
2. Implementing a `getContext()` wrapper that returns the context instance (again if fetched from $environment validation/exception is already done., only thing necessary is adding the full NameSpace to the class name)
3. on methods that use alt. contexts, call $this->getContext( name )->callSomeMethod();